File: 3rd_task/3rd task.cs
using System;

class Program{
    static void Main(){
        Console.Write("Введите количество чисел: ");
        int n = int.Parse(Console.ReadLine());

        int count = 0;
        Console.WriteLine("Введите {0} чисел: через Enter ", n);
        for (int i = 0; i < n; i++){
            int num = int.Parse(Console.ReadLine());
            if (IsPrime(num)){
                count++;
            }
        }

        Console.WriteLine("Количество простых чисел: " + count);
    }

    static bool IsPrime(int num){
        if (num <= 1){
            return false;
        }
        for (int i = 2; i * i <= num; i++){
            if (num % i == 0){
                return false;
            }
        }
        return true;
    }
}

File: 3rd_task/3rd task.rb
def is_prime(num)
    return false if num <= 1
    (2..Math.sqrt(num)).each do |i|
      return false if num % i == 0
    end
    true
  end
  
  puts "Введите количество чисел: "
  n = gets.to_i
  
  count = 0
  puts "Введите число через Enter: "
  n.times do
    num = gets.to_i
    count += 1 if is_prime(num)
  end
  
  puts "Количество простых чисел: #{count}"
  